### PR TITLE
[Synapse] Add support for KQL script

### PR DESCRIPTION
**Pull request by wonner**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1076
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by isra-fel — CHANGES_REQUESTED** [review]
Please note that I updated the base branch to `release-2021-12-07` as we have begun the release process. Also, there are 2 conflicts after another Synapse PR is merged.

File: src/Accounts/Authentication/Utilities/CustomAssemblyResolver.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static class CustomAssemblyResolver
         private static IDictionary<string, Version> NetFxPreloadAssemblies =
             new Dictionary<string, Version>(StringComparer.InvariantCultureIgnoreCase)
             {
-                {"Azure.Core", new Version("1.20.0.0")},
+                {"Azure.Core", new Version("1.21.0.0")},
```
[the remark this answers is not in the document]
Comment (wonner) [reply src/Accounts/Authentication/Utilities/CustomAssemblyResolver.cs:27]: Done.

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**